jaxley/modules/base.py

```diff
@@ -392,7 +392,11 @@ def _reformat_index(self, idx: Any, dtype: type = int) -> np.ndarray:
             array of indices of shape (N,)"""
         if is_str_all(idx):  # also asserts that the only allowed str == "all"
             return idx
-        np_dtype = np.dtype(int).type if dtype is int else np.dtype(float).type
+
+        if isinstance(idx, np.ndarray) and np.issubdtype(idx.dtype, np.number):
```

Comment:
I just noticed, we could include a few more specific test cases for int/float np dtypes [here](https://github.com/jaxleyverse/jaxley/blob/8c7750a819469d7ef019e5385bde1b05133ee17b/tests/test_viewing.py#L292).

Reply:
I wasn't sure what test cases to add to check the float `np.float` dtypes. From the function description, it seems floats can be used for calling in the `loc` function, but I don't fully understand it.

Other than that I did add a few test cases to validate if everything is running with any integer dtypes.